test/integration/framework/test_server.go

```diff
@@ -126,11 +126,18 @@ func StartTestServer(t *testing.T, stopCh <-chan struct{}, setup TestServerSetup
 	if err != nil {
 		t.Fatal(err)
 	}
+	// use a channel to hand off the error
+	errs := make(chan error, 1)
 	go func() {
 		if err := kubeAPIServer.GenericAPIServer.PrepareRun().Run(stopCh); err != nil {
-			t.Fatal(err)
+			errs <- err
 		}
 	}()
+	// wait for it
+	err = <-errs
```

Comment:
blocking here until the server stops defeats the purpose of starting it in a goroutine

Reply:
Will change back to t.Error